### PR TITLE
Dim Only Query: validate schema required filters

### PR DESCRIPTION
**Pull request by pranavbhole**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: core/src/test/scala/com/yahoo/maha/core/RequestModelTest.scala [hunk 1]
```diff
@@ -5819,5 +5819,29 @@ class RequestModelTest extends FunSuite with Matchers {
     println(res.failed.get.getMessage)
     res.failed.get.getMessage should startWith (s"requirement failed: ERROR_CODE:10007 (Ad Format Name) can't be used with advertiser schema in publicFact cube")
   }
+
+  test ("Dim Only query Schema required filter validation") {
```
Comment (panditsurabhi) [inline core/src/test/scala/com/yahoo/maha/core/RequestModelTest.scala:5823]: Can we add a test to ensure it doesn't fail when the Advertiser ID filter is present?

Comment (pranavbhole) [reply core/src/test/scala/com/yahoo/maha/core/RequestModelTest.scala:5823]: working on it,

Comment (pranavbhole) [reply core/src/test/scala/com/yahoo/maha/core/RequestModelTest.scala:5823]: added one positive case, found that we have many positive cases all over test classes, shouldnt be problem.

**Comment by pranavbhole** [comment]
https://github.com/yahoo/maha/issues/649

**Pull request merged**